generators/server/templates/src/main/java/package/config/ElasticsearchConfiguration.java.ejs

```diff
@@ -0,0 +1,72 @@
+<%#
+ Copyright 2013-2018 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see https://www.jhipster.tech/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+package <%=packageName%>.config;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.vanroy.springdata.jest.JestElasticsearchTemplate;
+import com.github.vanroy.springdata.jest.mapper.DefaultJestResultsMapper;
+import io.searchbox.client.JestClient;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+import org.springframework.data.elasticsearch.core.ElasticsearchOperations;
+import org.springframework.data.elasticsearch.core.EntityMapper;
+import org.springframework.data.elasticsearch.core.convert.ElasticsearchConverter;
+import org.springframework.data.elasticsearch.core.mapping.SimpleElasticsearchMappingContext;
+import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder;
+
+import java.io.IOException;
+
+@Configuration
+public class ElasticsearchConfiguration {
+
+    @Bean
+    @Primary
+    public ElasticsearchOperations elasticsearchTemplate(final JestClient jestClient,
+                                                         final ElasticsearchConverter elasticsearchConverter,
+                                                         final SimpleElasticsearchMappingContext simpleElasticsearchMappingContext,
+                                                         Jackson2ObjectMapperBuilder jackson2ObjectMapperBuilder) {
```

Comment:
What about this one?
@Configuration
@EnableConfigurationProperties(ElasticsearchProperties.class)
public class ElasticsearchConfiguration {
`private ObjectMapper mapper;
    private final JavaTimeModule module;

    public ElasticsearchConfiguration(ObjectMapper mapper, JavaTimeModule module) {
        this.mapper = mapper;
        this.module = module;
    }

    @Bean
    public EntityMapper getEntityMapper() {
        return new CustomEntityMapper(mapper);
    }

    @Bean
    public ElasticsearchOperations elasticsearchTemplate(Client client, EntityMapper mapper) {
        return new JestElasticsearchTemplate(
            jestClient,
            elasticsearchConverter,
            new DefaultJestResultsMapper(simpleElasticsearchMappingContext, mapper));
    }

    public class CustomEntityMapper implements EntityMapper {

        private ObjectMapper objectMapper;

        public CustomEntityMapper(ObjectMapper objectMapper) {
            this.objectMapper = objectMapper;
            objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
            objectMapper.configure(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY, true);
        }

        @Override
        public String mapToString(Object object) throws IOException {
            return new String(objectMapper.writeValueAsString(object));
        }

        @Override
        public <T> T mapToObject(String source, Class<T> clazz) throws IOException {
            return objectMapper.readValue(source, clazz);
        }
    }
}`
ObjectMapper is a Singleton instantiated by spring, and EntityMapper shall be a bean (not a prototype)


Also, in the JacksonConfiguration, you can add these two beans to parse date and time:

`/**
     * Support for Java date and time API.
     * @return the corresponding Jackson module.
     */
    @Bean
    public JavaTimeModule javaTimeModule() {
        return new JavaTimeModule();
    }

    @Bean
    public Jdk8Module jdk8TimeModule() {
        return new Jdk8Module();
    }
`

This should be sufficient to handle date and time.
However, It didn't succeed for Instants on my side, so I annotated my field with 
` @JsonDeserialize(using = WidgetInstantDeserializer.class)
    private Instant timestamp;`

And made the deserializer:
`public class WidgetInstantDeserializer extends JsonDeserializer<Instant> {


    @Override
    public Instant deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
        String fieldName = p.nextFieldName();
        int nano = -1;
        if (fieldName.equals("nano")) {
            nano = p.nextIntValue(-1);
        }
        String epoch = p.nextFieldName();
        int epochVal = -1;
        if (epoch.equals("epochSecond")) {
            epochVal = p.nextIntValue(-1);
        }
        Instant i = Instant.ofEpochSecond(epochVal, nano);
        return i;
    }
}`

WDYT? Do you want to make it or that I PR?

Reply:
@Tcharl Sure It should be better to use the new [jackson-modules-java8](https://github.com/FasterXML/jackson-modules-java8).

 Your issue with the Instant is really strange. How is provided the Instant value and stored in ES ?
If come from the frontend (Moment type) you don't have an issue with the @RestController who use normaly the same ObjectMapper ?

If it's in nanoseconds format maybe you can try to add the deserialisation feature READ_DATE_TIMESTAMPS_AS_NANOSECONDS (see [TestInstantSerialization.java](https://github.com/FasterXML/jackson-modules-java8/blob/b45e632dbf4911c49cf33d2e8da5eb31113d1d75/datetime/src/test/java/com/fasterxml/jackson/datatype/jsr310/TestInstantSerialization.java) )

Reply:
The actual json format stored in ES is an Object:
{ "mykey": {"nano": anInt, "epoch": anotherInt }} which is weird because the Jdk8 module isn't able to deserialize it (see the deserializer trick).

I will try the option and provide feedback Asap, because I don't like parsing json manually...